common/src/main/java/com/wynntils/screens/partymanagement/PartyManagementScreen.java

```diff
@@ -0,0 +1,426 @@
+        Scoreboard scoreboard = McUtils.mc().level.getScoreboard();
+        Set<String> onlineUsers = new HashSet<>(scoreboard.getTeamNames());
+        onlineUsers.retainAll(Models.Friends.getFriends());
+
+        List<String> suggestedPlayers = new ArrayList<>(onlineUsers);
+        suggestedPlayers.removeAll(Models.Party.getPartyMembers()); // No need to suggest party members
+        suggestedPlayers.sort(String.CASE_INSENSITIVE_ORDER);
+
+        suggestedPlayersWidgets.clear();
+        for (int i = 0; i < suggestedPlayers.size(); i++) {
+            String playerName = suggestedPlayers.get(i);
+            if (playerName == null) continue;
+
+            suggestedPlayersWidgets.add(new SuggestionPlayer(
+                    this.width / 2 + 204, this.height / 2 - 125 + i * 20 - 10, 200, 20, playerName));
+        }
+    }
+
+    @SubscribeEvent
+    public void onFriendJoin(FriendConnectionEvent.Join e) {
+        reloadSuggestedPlayersWidgets();
+    }
+
+    @SubscribeEvent
+    public void onFriendLeave(FriendConnectionEvent.Leave e) {
+        // Delay the reload to allow the scoreboard to update
+        Managers.TickScheduler.scheduleLater(this::reloadSuggestedPlayersWidgets, 5);
+    }
+
+    /**
+     * Reloads the list of party members widgets
+     * <p>
+     * This should be called when the party list is updated or when the refresh button is pressed
+     */
+    private void reloadMembersWidgets() {
+        partyMembersWidgets.clear();
+        List<String> partyMembers = new ArrayList<>(Models.Party.getPartyMembers());
+        for (int i = 0; i < partyMembers.size(); i++) {
+            String playerName = partyMembers.get(i);
+            if (playerName == null) continue;
+
+            partyMembersWidgets.add(new PartyPlayer(
+                    this.width / 2 - X_START + 4,
+                    this.height / 2 - 125 + i * 20 - 10,
+                    200,
+                    20,
+                    playerName,
+                    offlineMembers.contains(playerName)));
+        }
+    }
+
+    @SubscribeEvent
+    public void onPartyUpdate(RelationsUpdateEvent.PartyList e) {
+        reloadMembersWidgets();
+        reloadSuggestedPlayersWidgets(); // Reload because we don't want to suggest party members
+    }
+
+    private void inviteFromField() {
+        // Remove all except commas, semicolons, whitespaces, and characters possible in name
+        String fieldText =
+                INVITE_REPLACER.matcher(inviteInput.getTextBoxInput()).replaceAll("");
+        fieldText = COMMA_REPLACER.matcher(fieldText).replaceAll(","); // semicolons and spaces to comma
+        if (fieldText.isBlank()) return;
+
+        if (!Models.Party.isPartying()) {
+            createParty();
+        }
+
+        Set<String> toInvite = new HashSet<>(List.of(fieldText.split(",")));
+        toInvite.removeAll(Models.Party.getPartyMembers());
+        toInvite.forEach(playerName -> McUtils.sendCommand("party invite " + playerName));
```

Comment:
Move to model (keep the foreach etc, but command knowledge is not a screen's duty)